history/management/commands/compare_perf.py

```diff
@@ -1,58 +1,59 @@
 import datetime
 from django.db import transaction
+
+
 class Command(BaseCommand):
 
     help = 'compares market perf vs nn_perf'
 
     def handle(self, *args, **options):
-        #setup
+        # setup
         buffer_between_prediction_and_this_script_mins = datetime.datetime.now().minute % 10
-        granularity_mins = 10 #TODO: change me when granularity changes
+        granularity_mins = 10  # TODO: change me when granularity changes
         ticker = 'BTC_ETH'
 
-        #get data
-        date_of_timerange_we_care_about_predictions_start = datetime.datetime.now() - datetime.timedelta(seconds=((granularity_mins)*60+(60*(1+buffer_between_prediction_and_this_script_mins))))
-        date_of_timerange_we_care_about_predictions_end = datetime.datetime.now() - datetime.timedelta(seconds=((granularity_mins)*60))
-        tr_timerange_end = TradeRecommendation.objects.filter(symbol=ticker,created_on__gte=date_of_timerange_we_care_about_predictions_start,created_on__lte=date_of_timerange_we_care_about_predictions_end).order_by('-created_on').first().created_on
+        # get data
+        date_of_timerange_we_care_about_predictions_start = datetime.datetime.now() - datetime.timedelta(seconds=((granularity_mins) * 60 + (60 * (1 + buffer_between_prediction_and_this_script_mins))))
+        date_of_timerange_we_care_about_predictions_end = datetime.datetime.now() - datetime.timedelta(seconds=((granularity_mins) * 60))
+        tr_timerange_end = TradeRecommendation.objects.filter(symbol=ticker, created_on__gte=date_of_timerange_we_care_about_predictions_start, created_on__lte=date_of_timerange_we_care_about_predictions_end).order_by('-created_on').first().created_on
         tr_timerange_start = tr_timerange_end - datetime.timedelta(seconds=120)
         price_timerange_start = tr_timerange_end
-        price_timerange_end = tr_timerange_end + datetime.timedelta(seconds=(granularity_mins*60))
-        trs = TradeRecommendation.objects.filter(created_on__gte=tr_timerange_start,created_on__lte=tr_timerange_end)
-        price_now = Price.objects.filter(symbol=ticker,created_on__lte=price_timerange_end).order_by('-created_on').first().price
-        price_then = Price.objects.filter(symbol=ticker,created_on__lte=price_timerange_start).order_by('-created_on').first().price
-
-        ## nn attributes
-        pct_buy = round(1.0 * sum(tr.recommendation == 'BUY' for tr in trs) / len(trs),2)
-        pct_sell = round(1.0 * sum(tr.recommendation == 'SELL' for tr in trs) / len(trs),2)
-        pct_hold = round(1.0 * sum(tr.recommendation == 'HOLD' for tr in trs) / len(trs),2)
+        price_timerange_end = tr_timerange_end + datetime.timedelta(seconds=(granularity_mins * 60))
+        trs = TradeRecommendation.objects.filter(created_on__gte=tr_timerange_start, created_on__lte=tr_timerange_end)
+        price_now = Price.objects.filter(symbol=ticker, created_on__lte=price_timerange_end).order_by('-created_on').first().price
```

Comment:
Similair as above.
How about we check if that kind of ```Price``` exists in the first place. If it doesn't we display an appropriate message. This could be done at the beginning of the management command.
This will throw an AttributeError if such ```Price``` does not exist.